astropy/time/core.py

```diff
@@ -2313,7 +2321,16 @@ def __init__(self, val, val2=None, format=None, scale=None, copy=False):
                 self._set_scale(scale)
         else:
             if format is None:
-                format = 'datetime' if isinstance(val, timedelta) else 'jd'
+                if isinstance(val, timedelta):
+                    format = 'datetime'
+                else:
+                    if getattr(val, 'unit', None) is None:
+                        warn(
```

Comment:
Nitpick: why not have things start on this line? And closing parenthesis after the warning? (I see the point of blacken, but do not always like the result much -- too C-inspired for my taste, and too many lines means less is visible in a single editor view)

Reply:
I factored out the code into a `_get_format` method.